### DISCUSSION TITLE
Using the AWS Advanced JDBC Wrapper with HikariCP and MySQL

### DISCUSSION THREAD

**Discussion by jpollard-cs** [aws/aws-advanced-jdbc-wrapper#1215] [Q&A]
Hello I'm trying to better understand a few things with regards to configuring the AWS Advanced JDBC Wrapper with HikariCP and MySQL. I notice [here](https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingSpringJdbcTemplateMySQLExample.java#L101) that the example is using `ds.addDataSourceProperty("targetDataSourceClassName", "com.mysql.cj.jdbc.MysqlDataSource");`. The [HikariCP docs](https://github.com/brettwooldridge/HikariCP/blob/dev/README.md#popular-datasource-class-names), however, indicate "The MySQL DataSource is known to be broken with respect to network timeout support. Use jdbcUrl configuration instead.". From what I understand I believe these network timeout issues should still be relevant when using the `AwsWrapperDataSource` with `MysqlDataSource` as the underlying `targetDataSourceClassName`? One thing I don't entirely understand is if this will have a significant impact on how physical connections are replaced in the event of a failover (for context our typical plugin configuration would be `failover2,efm2,auroraConnectionTracker`)? One of my concerns here is whether or not HikariCP will still maintain control over new connection creation? My underlying concern here is ensuring the preservation of default settings the pool has been instructed to use when establishing a new connection. I know some connection settings are tracked, but I'm not particularly concerned with this as we do not typically change connection settings after connections are established (e.g. if we have a special use case that requires a non-standard transaction isolation level we typically recommend establishing a pool for this use case where all connections have this isolation level setting). Any guidance here would be much appreciated!

**Comment by erikrant-p** [comment]
Why was this issue closed? I am also concerned that docs [here](https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/DataSource.md) it is suggested that "When using the AwsWrapperDataSource you must specify the dataSourceClassName, and the HikariDataSource.setJdbcUrl method should not be used.". HikariCP docs suggest [here](https://github.com/brettwooldridge/HikariCP?tab=readme-ov-file#popular-datasource-class-names) that the opposite should be done by setting jdbcUrl due to MySQL DataSource's broken network timeout support
